static/js/util/dom.test.ts

```diff
@@ -0,0 +1,103 @@
+import { mockMatchMedia } from "../test_util"
+import { scrollToElement } from "./dom"
+
+describe("scrollToError", () => {
+  const stringToHtml = (string: string): HTMLElement => {
+    const template = document.createElement("template")
+    template.innerHTML = string.trim()
+    const node = template.content.firstChild
+    if (node instanceof HTMLElement) return node
+    throw new Error("Expected an HTMLElement")
+  }
+  const getTestPage = () => {
+    const page = stringToHtml(`
```

Comment:
this is totally fine as is, but you can also do JSX and use `ReactDOMServer.renderToString` for stuff like this